src/cpp/session/modules/environment/SessionEnvironment.cpp

```diff
@@ -720,25 +772,56 @@ Error setContextDepth(boost::shared_ptr<int> pContextDepth,
 
 Error getEnvironmentState(boost::shared_ptr<int> pContextDepth,
                           boost::shared_ptr<LineDebugState> pLineDebugState,
-                          const json::JsonRpcRequest&,
+                          const json::JsonRpcRequest& request,
                           json::JsonRpcResponse* pResponse)
 {
-   pResponse->setResult(commonEnvironmentStateData(*pContextDepth,
-                                                   true, // include contents
-                                                   pLineDebugState.get()));
+   std::string language = "R";
+   std::string environment = "R_GlobalEnv";
+   Error error = json::readParams(request.params, &language, &environment);
+   if (error)
+      LOG_ERROR(error);
+   
+   json::Object jsonState;
+   
+   if (language == "Python")
+   {
+      jsonState = pythonEnvironmentStateData(environment);
+   }
+   else
+   {
+      jsonState = commonEnvironmentStateData(
+               *pContextDepth,
+               true,
+               pLineDebugState.get());
+   }
+   
+   pResponse->setResult(jsonState);
    return Success();
 }
 
-void onDetectChanges(module_context::ChangeSource source)
+void onDetectChanges(module_context::ChangeSource /* source */)
 {
    // Prevent recursive calls to this function
    DROP_RECURSIVE_CALLS;
 
+   // Check for Python changes.
+   if (s_environmentLanguage == "Python" && !s_monitoredPythonModule.empty())
+   {
+      Error error =
+            r::exec::RFunction(".rs.reticulate.detectChanges")
+            .addParam(s_monitoredPythonModule)
+            .call();
+      
+      if (error)
+         LOG_ERROR(error);
+   }
+   
    // Ignore if not monitoring
    if (!s_monitoring)
```

Comment:
Does this imply that the Python environment pane does not respect the manual/auto refresh setting? (Should it?)

Reply:
You're right! I've made that change.